sympy/polys/domains/mpelements.py

```diff
@@ -57,8 +57,12 @@ def __init__(ctx, prec=53, dps=None, tol=None):
         ctx.mpc = ComplexElement
         ctx.mpf._ctxdata = [ctx.mpf, new, ctx._prec_rounding]
         ctx.mpc._ctxdata = [ctx.mpc, new, ctx._prec_rounding]
-        ctx.mpf.context = ctx
-        ctx.mpc.context = ctx
+
+        if field == 'RR':
```

Comment:
It would be more efficient to have a flag like `real`. Then this would be `if real:`.